render/src/pixel_bender.rs

```diff
@@ -564,6 +567,11 @@ fn read_op<R: Read>(
                 _ => unreachable!(),
             }
         }
+        Opcode::Loop => {
+            let mut unknown = vec![0u8; 23];
```

Comment:
Am I missing something, or this doesn't have to be a Vec?

Reply:
I store a `Vec` in the `Operation::Loop` enum variant, to make it cheaper to move.

Reply:
Cheaper to move than what? It feels like this could just be a slice

Reply:
Storing an array in `Operation::Loop` will make the entire enum larger. Since we're using a vec, there's no reason to start out with an array in this method and then copy it to a vec.

Reply:
Doesn't a Vec have a size of 24 bytes? An array of 23 u8s should by 23 bytes, if I'm not mistaken

Reply:
Forgive me if I'm missing something

Reply:
On 32-bit platforms like wasm, it's 12 bytes.

Reply:
Right, my apologies then. But wouldn't a boxed array still save space?

Reply:
Good catch - I've changed it to `Box<[u8]>`